corehq/apps/userreports/reports/view.py

```diff
@@ -78,12 +93,28 @@ def has_permissions(self, domain, user):
         return True
 
     def get_context_data(self, **kwargs):
-        return {
+        context = {
             'domain': self.domain,
             'report': self,
             'filter_context': self.filter_context,
-            'url': reverse(self.slug, args=[self.domain, self.report_config_id]),
-            'headers': self.headers,
+            'url': self.url,
+            'headers': self.headers
+        }
+        context.update(self.saved_report_context_data)
+        return context
+
+    @property
+    def saved_report_context_data(self):
+        current_config_id = self.request.GET.get('config_id')
+        return {
+            'report_configs': ReportConfig.by_domain_and_owner(
+                self.domain, self.request.couch_user._id, report_slug=self.slug
+            ),
```

Comment:
maybe this has to be here in which case that's fine, but it's going to slow down every UCR page.